### ISSUE TITLE
ROOT tag file contains hard coded paths

### ISSUE THREAD

**Issue by ferdymercury** [root-project/root#10059]
- [x] Checked for duplicates

### Describe the bug
The ROOT doxygen tag files posted in https://root.cern/reference/ contain full hard coded paths that are not available from outside users.

### Expected behavior
No hard-coded paths are shown there.

### To Reproduce
- Download the ROOT.tag from the link above
- grep sftnight ROOT.tag

### Additional context
https://github.com/doxygen/doxygen/issues/9185

**Comment by couet** [comment]
in Doxyfile we have:
```
STRIP_FROM_PATH        = $(DOXYGEN_STRIP_PATH)
```
And `DOXYGEN_STRIP_PATH` is defined in `Makefile` the following way:
```
DOXYGEN_STRIP_PATH := $(shell cd $(PWD)/../..; pwd)
```
Is that ok ? looks fine to me. No ?

**Comment by ferdymercury** [comment]
Yes, but that's the ROOT trunk source directory.

Somehow, in ROOT.tag we are also including files from the [rootspi](https://github.com/root-project/rootspi) repository, and that's maybe not in the STRIP_PATH right now? (Probably these are generated on the fly by the script or something like that?)

If not, maybe this is a bug in doxygen. We could report it if we find a minimal reproducer.

**Comment by couet** [comment]
I see it is `/home/sftnight/build/workspace/root-makedoc-master/rootspi/rdoc/master_TMP/pyzdoc/` . So the `pyzdoc` which I guess should e also put in `STRIP_FROM_PATH ` somehow ?

**Comment by couet** [comment]
Yes that's it because that folder is added in the list of folders to be scanned by doxygen at the end of "makeinput.sh`

**Comment by ferdymercury** [comment]
Yes. STRIPfrompath accepts an array of paths (space separated), not just a single path.

**Comment by ferdymercury** [comment]
I filed a bug report in doxygen https://github.com/doxygen/doxygen/issues/9221

**Comment by couet** [comment]
have you tried `STRIP_FROM_PATH ` ? Why do you think there is a bug ?

**Comment by ferdymercury** [comment]
yes, I tried and the path is still there

**Comment by ferdymercury** [comment]
by the way, do you know why the root-makedoc-master building is not running through since one month ? (https://lcgapp-services.cern.ch/root-jenkins/view/ROOT/job/root-makedoc-master/)
[Last stable build (#2613 [5d12207b]), 1 mo 7 days ago](https://lcgapp-services.cern.ch/root-jenkins/view/ROOT/job/root-makedoc-master/lastStableBuild/)

**Comment by couet** [comment]
It runs nut it is marked as failing when a tutorial fails.

**Comment by ferdymercury** [comment]
> I filed a bug report in doxygen [doxygen/doxygen#9221](https://github.com/doxygen/doxygen/issues/9221)

A fix has been proposed. (PR not yet merged)

**Comment by couet** [comment]
Let me know when it will be in and I will install the new version.

**Comment by ferdymercury** [comment]
@couet the PR of doxygen has been merged into master. So now, I would suggest to add the missing pyzdoc folder to DOXYGEN_STRIP_PATH in the Makefile, as dicussed above, and upgrade doxygen on the build machine.

**Comment by couet** [comment]
@ferdymercury sorry for the late reply. I was off last week. I'll look asap.

**Comment by ferdymercury** [comment]
The feature is now on release 1.9.6

**Comment by dpiparo** [comment]
@couet , ping.

**Comment by couet** [comment]
_The feature is now on release 1.9.6_

@ferdymercury I guess we can close this one since the doc is now build with a github action using doxygen 1.10.0

**Comment by ferdymercury** [comment]
Thanks @couet for the feedback!

The issue is still there, but I think it's very easy to solve.

If you grep ROOT.tag, you will see that there is the hardcoded path  `/github/home/master/pyzdoc/..`..

To close this issue, we would need to add DOXYGEN_PYZDOC_PATH to the STRIP_PATH variable, or alternatively just DOXYGEN_OUTPUT_DIRECTORY

**Comment by couet** [comment]
Ok I see this:

```
% grep pyzdoc ROOT.tag | grep master
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
    <path>/github/home/master/pyzdoc/</path>
```
and I see that STRIP_PATH appears in:
```
% grep STRIP_PATH *
Doxyfile:STRIP_FROM_PATH        = $(DOXYGEN_STRIP_PATH)
Makefile:export DOXYGEN_STRIP_PATH := $(shell cd $(PWD)/../..; pwd)
Makefile:export DOXYGEN_INCLUDE_PATH := $(shell find $(DOXYGEN_STRIP_PATH) -type d -name dictpch -prune -o -type d -name inc)
```
What modification would you suggest exactly ?

**Comment by ferdymercury** [comment]
/github/home/master/ should not appear in the tag. To do that, stripfrompath should be expanded to include doxygenoutputdir

**Comment by couet** [comment]
Thanks for your reply. I modified the Makefile this way:
```
export DOXYGEN_STRIP_PATH := $(shell cd $(PWD)/../..; pwd),$(DOXYGEN_OUTPUT_DIRECTORY)
```
ie using comme "`,`" as separator but I am a bit unsure of it. May bet it is ":" like for the `$PATH` variable ? I looked a bit on the web and did not find any example of `DOXYGEN_STRIP_PATH` with several pathes in it. Do you have some idea/guess abot it ?

**Comment by ferdymercury** [comment]
> ie using comme "`,`" as separator but I am a bit unsure of it. May bet it is ":" like for the `$PATH` variable ? I looked a bit on the web and did not find any example of `DOXYGEN_STRIP_PATH` with several pathes in it. Do you have some idea/guess abot it ?

I think the format is space-separated.
And I think we have to maybe append that AFTER creating the variable DOXYGEN_INCLUDE_PATH ?
